docs/source/gateway/index.rst

```diff
@@ -1,6 +1,782 @@
+            openai_api_version: "2023-05-15"
+
+
+.. note::
+
+    Azure OpenAI has disctinct features as compared with the direct OpenAI service. For an overview, please see `the comparison documentation <https://learn.microsoft.com/en-gb/azure/cognitive-services/openai/how-to/switching-endpoints>`_.
+
+For specifying an API key, there are three options:
+
+1. (Preferred) Use an environment variable to store the API key and reference it in the YAML configuration file. This is denoted by a ``$`` symbol before the name of the environment variable.
+2. (Preferred) Define the API key in a file and reference the location of that key-bearing file within the YAML configuration file.
+3. Directly include it in the YAML configuration file.
+
+.. important::
+
+    The use of environment variables or file-based keys is recommended for better security practices. If the API key is directly included in the configuration file, it should be ensured that the file is securely stored and appropriately access controlled.
+    Please ensure that the configuration file is stored in a secure location as it contains sensitive API keys.
+
+Querying the AI Gateway
+=======================
+
+Once the MLflow AI Gateway server has been configured and started, it is ready to receive traffic from users.
+
+AI Gateway Server
+-----------------
+The Mlflow AI Gateway server has endpoints that are not accessible via the fluent or client APIs. These endpoints are used primarily for service management, exploration, and configuration validation.
+
+FastAPI Documentation ("/docs")
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+FastAPI, the framework used for building the MLflow AI Gateway, provides an automatic interactive API
+documentation interface, which is accessible at the "/docs" endpoint (e.g., ``http://my.gateway:9000/docs``).
+This interactive interface is very handy for exploring and testing the available API endpoints.
+
+As a convenience, accessing the root URL (e.g., ``http://my.gateway:9000``) redirects to this "/docs" endpoint.
+
+Examples of Post Requests
+~~~~~~~~~~~~~~~~~~~~~~~~~
+You can use the POST request to send a query to a specific route.
+To send a query to a specific route, append the route name to the routes endpoint, and include the
+data to be sent in the body of the request. The structure of this data will depend on the specific model the route is configured for.
+
+For instance, to send a query to the completions route, you might use the following command:
+
+.. code-block:: bash
+
+    curl -X POST -H "Content-Type: application/json" \
+      -d '{"prompt": "It is a truth universally acknowledged"}' \
+      http://my.gateway:9000/gateway/routes/completions/invocations
+
+This will return a JSON object with the response from the completions model, which is usually the continuation of the text provided as a prompt.
+
+**Note:** Please remember to replace ``http://my.gateway:9000`` with the URL of your actual Gateway Server.
+
+MLflow Python Client APIs
```

Comment:
Can we add an "MLflow Models" section between this section and "REST APIs" that shows users how to query the gateway within an MLflow custom pyfunc model and how to set the gateway URI at deployment time using the MLFLOW_GATEWAY_URI environment variable? Here's an example from Databricks: https://docs.google.com/document/d/1HP5Kblja6Z5wM9mEeRVY6ZExOX7B59U_ZCrTdip99jg/edit#heading=h.yco7e24lxaen